### PR TITLE
New Resource `azurerm_network_manager_management_group_connection`

### PR DESCRIPTION
**Pull request by teowa**
Notes:

- This resource is the `Virtual network manager hub connection`(with management group) metioned in [cross-tenant-connection](https://learn.microsoft.com/en-us/azure/virtual-network-manager/concept-cross-tenant#cross-tenant-connection) doc,  testcases of this resource uses a single tenant due to test limitations.
- swagger: https://github.com/Azure/azure-rest-api-specs/blob/df77748f7dff3fe12c13b7262e9b307c719e6918/specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/networkManagerConnection.json#L190

### CODE REVIEW

File: internal/services/network/client/client.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type Client struct {
 	IPGroupsClient                         *network.IPGroupsClient
 	LocalNetworkGatewaysClient             *network.LocalNetworkGatewaysClient
 	ManagersClient                         *network.ManagersClient
+	ManagerManagementGrpConnectionsClient  *network.ManagementGroupNetworkManagerConnectionsClient
```
Comment (tombuildsstuff) [inline internal/services/network/client/client.go:29]: Grp -> Group

File: internal/services/network/network_manager_management_group_connection_resource.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+		"management_group_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: managementValidate.ManagementGroupID,
+		},
+
+		"network_manager_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_management_group_connection_resource.go:61]: shouldn't this be ForceNew?

File: internal/services/network/network_manager_management_group_connection_resource.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				if model.Description != "" {
+					properties.Description = &model.Description
+				} else {
+					properties.Description = nil
+				}
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_management_group_connection_resource.go:160]: why do we need to explicitly nil this? these fields are `omitempty` so this won't be sent..

File: internal/services/network/network_manager_management_group_connection_resource.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+			}
+
+			if metadata.ResourceData.HasChange("network_manager_id") {
+				if model.NetworkManagerId != "" {
+					properties.NetworkManagerID = &model.NetworkManagerId
+				} else {
+					properties.NetworkManagerID = nil
+				}
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_management_group_connection_resource.go:168]: why do we need to explicitly nil this? these fields are `omitempty` so this won't be sent..

File: internal/services/network/network_manager_management_group_connection_resource.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+			}
+
+			existing.SystemData = nil
```
Comment (tombuildsstuff) [inline internal/services/network/network_manager_management_group_connection_resource.go:171]: why do we need to explicitly nil this? these fields are `omitempty` so this won't be sent..

File: internal/services/network/parse/network_manager_management_group_connection.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package parse
+
+// NOTE: this file is generated via 'go:generate' - manual changes will be overwritten
```
Comment (tombuildsstuff) [inline internal/services/network/parse/network_manager_management_group_connection.go:3]: this file has been hand-written..?

File: internal/services/network/parse/network_manager_management_group_connection_test.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
+package parse
+
+// NOTE: this file is generated via 'go:generate' - manual changes will be overwritten
```
Comment (tombuildsstuff) [inline internal/services/network/parse/network_manager_management_group_connection_test.go:3]: this file has been hand-written..?

File: internal/services/network/validate/network_manager_management_group_connection_id.go [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package validate
+
+// NOTE: this file is generated via 'go:generate' - manual changes will be overwritten
```
Comment (tombuildsstuff) [inline internal/services/network/validate/network_manager_management_group_connection_id.go:3]: this file has been hand-written..?

File: internal/services/network/validate/network_manager_management_group_connection_id_test.go [hunk 9]
```diff
@@ -0,0 +1,64 @@
+package validate
+
+// NOTE: this file is generated via 'go:generate' - manual changes will be overwritten
```
Comment (tombuildsstuff) [inline internal/services/network/validate/network_manager_management_group_connection_id_test.go:3]: this file has been hand-written..?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @teowa 

Thanks for this PR - I've taken a look through and left a first set of comments inline, unfortunately we're going to need to make a few changes tot his one but once those are done we should be able to take another look.

Thanks!

File: internal/services/network/network_manager_management_group_connection_resource.go (hunk 2 above)
Comment (teowa) [reply internal/services/network/network_manager_management_group_connection_resource.go:61]: The resource ID is like `/providers/Microsoft.Management/managementGroups/%s/providers/Microsoft.Network/networkManagerConnections/%s` , the `network_manager_id` is in the [property](https://github.com/Azure/azure-rest-api-specs/blob/df77748f7dff3fe12c13b7262e9b307c719e6918/specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/networkManagerConnection.json#L371-L389), which is not a `ForceNew`.

**Comment by teowa** [comment]
image

**Comment by teowa** [comment]
Provisioning two Network Manager using one subscription as scope leads to an error, to avoid test failure, temporary close this PR. 

Please review #19593 first.

**Comment by teowa** [comment]
image

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Pull request merged**